### ISSUE TITLE
update on useWarnWhenUnsavedChanges with react-router useBlock

### ISSUE THREAD

**Issue by mikel-k-khui** [marmelab/react-admin#9325]
**Is your feature request related to a problem? Please describe.**
The current implementation of useWarnWhenUnsavedChanges seem to fail silently due to react-router change.
https://marmelab.com/react-admin/Form.html#warnwhenunsavedchanges

**Describe the solution you'd like**
Is there plan to update this with the new `react-router` changes in 6.4.

**Comment by djhi** [comment]
Yes, we'll work on this when we can. As this is documented, we don't consider it a bug though:
- https://marmelab.com/react-admin/SimpleForm.html#warnwhenunsavedchanges
- https://marmelab.com/react-admin/SimpleForm.html#warnwhenunsavedchanges
- https://marmelab.com/react-admin/TabbedForm.html#warnwhenunsavedchanges

**Comment by djhi** [comment]
I believe this can now be considered fixed. Feel free to reopen if you think otherwise.

**Comment by slax57** [comment]
Related PR: https://github.com/marmelab/react-admin/pull/9657
